__tests__/app.test.js

```diff
@@ -144,3 +160,64 @@ describe("PATCH: /api/articles/:article_id should update corresponding article w
       });
   });
 });
+
+describe("GET: /api/articles should return an array of article objects sorted by descending date and takes a topic query", () => {
+  test("200: should return an array of objects that are sorted by date in descending order by default ", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              created_at: expect.any(String),
+              comment_count: expect.any(Number),
+              topic: expect.any(String),
+              votes: expect.any(Number),
+            })
+          );
+        });
+      });
+  });
+  test("200: should accept an optional topic query which filters out articles unrelated to topic", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body;
+        expect(articles).toBeSortedBy("created_at", { descending: true });
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              topic: "cats",
+            })
+          );
+        });
+      });
+  });
+  test("404: returns error if topic query does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=fish")
+      .expect(404)
+      .then((response) => {
+        const {
+          body: { msg },
+        } = response;
+        expect(msg).toBe("404: Topic not found");
+      });
+  });
+  test("returns empty array if topic exists but has no associated articles", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(200)
+      .then((response) => {
+        expect(response.body).toEqual([]);
+      });
+  });
+});
```

Comment:
I would think about adding tests to show that it accepts `sort_by` query, e.g. `?sort_by=votes` and accepts `order` query, e.g. `?order=desc`

If we are testing for this, you may want to  test for invalid `sort_by` query, e.g. `?sort_by=bananas` and invalid `order` query, e.g. `?order=bananas`